rr_iarrc/src/cone_detection/cone_detection_height.cpp

```diff
@@ -0,0 +1,336 @@
+    sure_bodies = cutSmall(sure_bodies, minimum_area_cut);
+
+    //Find Distance
+    frame = drawAndCalc(frame, sure_bodies);
+    ros::Time end = ros::Time::now();
+
+//	publish Images
+    publishMessage(pub, frame, "bgr8");
+    publishMessage(pub1, detected_bodies, "mono8");
+
+    pcl::PointCloud<pcl::PointXYZ> cone_cloud;
+    for (int i = 0; i < cone_points.size(); i++) {
+        drawCircle(cone_cloud, cone_points[i], real_cone_radius, 20);
+    }
+    sensor_msgs::PointCloud2 outmsg;
+    pcl::toROSMsg(cone_cloud, outmsg);
+    outmsg.header.frame_id = "base_footprint";
+    pub2.publish(outmsg);
+}
+
+int main(int argc, char** argv) {
+    ros::init(argc, argv, "coneDetectionHeight");
+
+    ros::NodeHandle nh;
+    ros::NodeHandle nhp("~");
+    std::string subscription_node;
+    nhp.param("orange_low_H", low_H, 5);
+    nhp.param("orange_high_H", high_H, 25);
+    nhp.param("orange_low_S", low_S, 140);
+    nhp.param("orange_low_V", low_V, 140);
+
+    nhp.param("canny_cut_min_threshold", canny_cut_min_threshold, 40);
+    nhp.param("percent_max_distance_transform", percent_max_distance_transform, 0.7);
+    nhp.param("minimum_area_cut", minimum_area_cut, 40);
+    
+    nhp.param("camera_focal_length_y", fy, 611.46);
+    nhp.param("real_cone_height", real_cone_height, .2286);
+    nhp.param("real_cone_radius", real_cone_radius, .07);
+
+    nhp.param("blockSky_height", blockSky_height, 220);
+    nhp.param("blockWheels_height", blockWheels_height, 200);
+    nhp.param("blockBumper_height", blockBumper_height, 200);
+
+    nhp.param("subscription_node", subscription_node, std::string("/camera/image_color_rect"));
+
+    loadCameraFOV(nh);
+    pub = nh.advertise<sensor_msgs::Image>("/cones/height/debug_coordinates", 1); //test publish of image
+    pub1 = nh.advertise<sensor_msgs::Image>("/cones/height/separated_bodies", 1);
+    pub2 = nh.advertise<sensor_msgs::PointCloud2>("/cones/height/pointcloud", 1); //test publish of image
+    auto img_real = nh.subscribe(subscription_node, 1, img_callback);
+
+    ros::spin();
+    return 0;
+}
+
+
+cv::Mat cutSmall(cv::Mat color_edges, int size_min) {
+    cv::Mat contours_color(color_edges.rows,color_edges.cols,CV_8UC1,cv::Scalar::all(0));
+    std::vector<std::vector<cv::Point>> contours;
+
+    cv::findContours(color_edges, contours,CV_RETR_CCOMP, CV_CHAIN_APPROX_SIMPLE);
+    for( int i = 0; i < contours.size(); i++ ) {
+        if (size_min < cv::arcLength(contours[i], false) ) {
+            cv::drawContours(contours_color, contours, i, cv::Scalar(255), CV_FILLED, 8);
+        }
+    }
+    return contours_color;
+}
+
+
+cv::Mat cutBodies(cv::Mat frame_cut, cv::Mat orange_found) {
+    Mat frame_gray, detected_edges, detected_bodies;
+
+    GaussianBlur(frame_cut, frame_cut, Size(3,3), 0, 0, BORDER_DEFAULT );
+    cv::cvtColor(frame_cut, frame_gray, cv::COLOR_BGR2GRAY );
+    bitwise_and(frame_gray, orange_found, frame_gray);
```

Comment:
This feels a confusing way to remove the environment from the orange found image. May want to make a note that is why you do it rather than using `cutEnvironment() `.